File: dataset/human/c/file_6682.c
#include <stdio.h>
#define BUF_SIZE 100
#define LONG_MIN -2147483646
#define LONG_MAX 2147483647

int main(void)
{
  char buf[100];
  long input, maxv = LONG_MIN, min = LONG_MAX;
  long linum;
  long diff = 0;

  fgets(buf, BUF_SIZE, stdin);
  sscanf(buf, "%ld", &linum);

  while(linum--){
    fgets(buf, BUF_SIZE, stdin);
    if(buf[0] == '\n') break;
    sscanf(buf, "%ld", &input);

    diff = input - min;
    if(maxv < diff){
      maxv = diff;
    }

    if(input < min){
      min = input;
    }

  }

  printf("%ld\n", maxv);

  return 0;
}

File: dataset/human/python/file_6682.py
import sys
s=input()
q=int(input())
a=[]
b=[]
c=0
for i in sys.stdin:
  if '1' in i.split()[0]:
    a,b=b,a
    c+=1
  else:
    if '1' in i.split()[1]:
      a.append(i.split()[2])
    else:
      b.append(i.split()[2])
if c%2==1:
  s=''.join(list(s)[::-1])
print(''.join(a[::-1])+s+''.join(b))
